src/libs/parsers/formatDates.ts

```diff
@@ -216,3 +218,18 @@ export function getTimeZonedDate(date: Date | string, timezone: string) {
   const utcDate = new Date(date)
   return utcToZonedTime(utcDate, timezone)
 }
+
+export function extractStockDates(offer: OfferResponse): string[] {
+  return offer.stocks.reduce<string[]>((accumulator, stock) => {
+    if (stock.beginningDatetime) {
+      return [...accumulator, stock.beginningDatetime]
+    }
+    return accumulator
+  }, [])
+}
```

Comment:
Tu pourrais faire plus simple :
```suggestion
export function extractStockDates(offer: OfferResponse): string[] {
  return offer.stocks.map(stock => stock.beginningDateTime).filter(Boolean)
}
```